Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1115 @@
+        V3D(0, 0, -10) /*source position*/, V3D(0, 0, 0) /*sample position*/,
+        V3D(0, 0, 10) /*bank position*/);
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+
+    // call saveinstrument taking test instrument as parameter
+    NexusGeometrySave::saveInstrument(instr, destinationFile,
+                                      DEFAULT_ROOT_PATH);
+
+    // test utility to check the output file
+    NexusFileReader tester(destinationFile);
+
+    // assert that inside a group with attribute NXentry, which as per the
+    // previous test we know to be the root group, there exists a group of
+    // NXclass NXinstrument.
+    TS_ASSERT(tester.parentNXgroupHasChildNXgroup(NX_ENTRY, NX_INSTRUMENT));
+  }
+
+  void test_NXclass_with_name_has_same_group_name_and_is_stored_in_dataset() {
+    // this test checks that when a name for some component in the instrument
+    // cache has been provided, saveInstrument will save the relevant group uder
+    // that name. this test is done for the done for the NXinstrument group. the
+    // name of the instrument will be manually set, then the test utility will
+    // try to open a group with that same same name, if such a group does not
+    // exist, a H5 group error is thrown. no such exception is expected to be
+    // thrown.
+
+    // RAII file resource for test file destination
+    ScopedFileHandle fileResource("check_instrument_name_test_file.nxs");
+    auto destinationFile = fileResource.fullPath();
+
+    // test instrument
+    auto instrument = ComponentCreationHelper::createMinimalInstrument(
+        V3D(0, 0, -10) /*source position*/, V3D(0, 0, 0) /*sample position*/,
+        V3D(0, 0, 10) /*bank position*/);
+
+    // set name of instrument
+    instrument->setName("test_instrument_name");
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+
+    // call saveInstrument passing the test instrument as parameter.
+    NexusGeometrySave::saveInstrument(instr, destinationFile,
+                                      DEFAULT_ROOT_PATH); // saves instrument
+
+    // test utility to check the output file.
+    NexusFileReader testUtility(destinationFile);
+
+    // full H5 path to the NXinstrument group
+    FullNXPath path = {DEFAULT_ROOT_PATH, "test_instrument_name"};
+
+    // assert no exception thrown on open of instrument group in file with
+    // manually set name.
+    TS_ASSERT_THROWS_NOTHING(testUtility.openfullH5Path(path));
+
+    // assert group is indeed NXinstrument.
+    TS_ASSERT(testUtility.hasNXAttributeInGroup(NX_INSTRUMENT, path));
+
+    // assert the dataset containing the instrument name has been correctly
+    // stored also.
+    TS_ASSERT(
+        testUtility.dataSetHasStrValue(NAME, "test_instrument_name", path));
+  }
+
+  void
+  test_NXclass_without_name_is_assigned_unique_default_name_for_each_group() {
+    // this test will try to save and unnameed instrument with multiple unnamed
```

Comment:
A couple of typos here:
"unnameed"
"vefiry"

Reply:
done